src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -880,22 +890,79 @@ QgsLegendModel::QgsLegendModel( QgsLayerTree *rootNode, QObject *parent )
 QVariant QgsLegendModel::data( const QModelIndex &index, int role ) const
 {
   // handle custom layer node labels
-  if ( QgsLayerTreeNode *node = index2node( index ) )
+
+  QgsLayerTreeNode *node = index2node( index );
+  QgsLayerTreeModelLegendNode *ltmln = index2legendNode( index ); // Possibly useless
```

Comment:
I'm a bit concerned about this comment. It's unclear who is the target audience and if there's a hidden risk and if yes what exactly.

Reply:
Ah yes, well I assumed at some point that the layertreemodllegendnode was getting its data method called at some point but during my tests, this was never the case. Only the feature itself gets called and I am forced to loop over the symbols. This is more of a safety net but it was always a nullptr in my tests, I can remove it if you want.

Reply:
If it serves no purpose, remove it.

Reply:
done